packages/number-field/src/NumberField.ts

```diff
@@ -685,6 +703,18 @@ export class NumberField extends TextfieldBase {
             );
             this.value = value;
         }
+        if (changes.has('step')) {
+            if (this.step) {
+                const digitsAfterDecimal =
+                    this.step != Math.floor(this.step)
+                        ? this.step.toString().split('.')[1].length
+                        : 0;
+                this._valueFormatter = new NumberFormatter(
+                    this.languageResolver.language,
+                    { maximumFractionDigits: digitsAfterDecimal }
+                );
```

Comment:
We shouldn't need to populate the cache here, if we evict it, the use of `this.valueFormatter` later will automatically populate it on demand.

Reply:
we are populating cache here because when ever the value of `step` changes we have to create a new instance of NumberFormatter. 
If instead of line 708-715 i just write `this.valueFormatter` this will not update the value of `decimalAfterDigits`.
For instance lets assume initially `this.step` was 0.1 and thus `decimalAfterDigits` was 1 and hence `this._valueFormatter` was initialised with `maximumFractionDigits: 1`. Now user updates `this.step` to 0.001, so `decimalAfterDigits` will become 3, now we have to again initialise `this._valueFormatter` with updated value of `decimalAfterDigits`. But if don't do so and just write `this.valueFormatter` instead of line 708-715, this will return `this._valueFormatter` with old value of `maximumFractionDigits`

Reply:
Done!